e2e-python/Jenkinsfile.template

```diff
@@ -0,0 +1,185 @@
+  podContainers: [
+      containerTemplate(
+        name: 'jnlp',
+        image: "${dockerRegistry}/ods/jenkins-agent-terraform-2306:@shared_library_ref@",
+        envVars: [
+          envVar(key: 'AWS_REGION', value: aws_region)
+        ],
+        alwaysPullImage: true,
+        args: '${computer.jnlpmac} ${computer.name}'
+      )
+    ],
+  branchToEnvironmentMapping: [
+    '*': 'dev',
+    // 'release/': 'test'
+  ]
+) { context ->
+    getEnvironment(context)
+    addVars2envJsonFile(context)
+    odsComponentStageInfrastructure(context, [cloudProvider: 'AWS'])
+
+    withEnv(["AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}",
+             "AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}"
+   ])
+   {
+      stage ("AWS Testing Preparation"){
+        generateTerraformOutputsFile()
+      }
+
+      def outputNames = stageGetNamesFromOutputs()
+      def aws_pipelineName = outputNames.aws_codepipeline_name
+      def bitbuckets3_name = outputNames.bitbuckets3_name
+      def results3_name = outputNames.results3_name
+
+      stage ("Publish Bitbucket Code To AWS"){
+        publishBitbucketCodeToAWS(context, bitbuckets3_name)
+      }
+
+      stage ("Run Tests"){
+        awsCodePipelineTrigger(context, aws_pipelineName)
+        awsCodePipelineWaitForExecution(context, aws_pipelineName)
+      }
+
+      stage ("Test Results"){
+        retrieveReportsFromAWS(context, results3_name)
+        archiveArtifacts artifacts: "build/test-results/test/**", allowEmptyArchive: true
+        junit(testResults:'build/test-results/test/*.xml', allowEmptyResults: true)
+        stash(name: "acceptance-test-reports-junit-xml-${context.componentId}-${context.buildNumber}", includes: "build/test-results/test/acceptance*junit.xml", allowEmpty: true)
+        stash(name: "installation-test-reports-junit-xml-${context.componentId}-${context.buildNumber}", includes: "build/test-results/test/installation*junit.xml", allowEmpty: true)
+        stash(name: "integration-test-reports-junit-xml-${context.componentId}-${context.buildNumber}", includes: "build/test-results/test/integration*junit.xml", allowEmpty: true)
+      }
+    }
+
+}
+
+def getEnvironment(def context){
+    sh "echo Get Environment Variables"
+    AWS_ACCESS_KEY_ID = sh(returnStdout: true, script:"oc get secret aws-access-key-id-${context.environment} --namespace ${context.cdProject} --output jsonpath='{.data.secrettext}' | base64 -d")
+    AWS_SECRET_ACCESS_KEY = sh(returnStdout: true, script:"oc get secret aws-secret-access-key-${context.environment} --namespace ${context.cdProject} --output jsonpath='{.data.secrettext}' | base64 -d")
+
+}
+
+
+def generateTerraformOutputsFile() {
+    sh 'terraform output -json > terraform_outputs.json'
+    sh 'cat terraform_outputs.json'
+}
+
+def stageGetNamesFromOutputs() {
+    def outputNames = [:]
+    def terraformOutputJson = readJSON file: 'terraform_outputs.json'
+    //def environmentVarsJson = readJSON file: env.auto.tfvars.json
```

Comment:
why this line is commented, if it is mean to be we should add an explanation above